paramiko/transport.py

```diff
@@ -310,6 +311,7 @@ def __init__(
         gss_kex=False,
         gss_deleg_creds=True,
         disabled_algorithms=None,
+        socket_timeout=10,
```

Comment:
I consider the default of `socket_timeout` should be `None`. It will change behavior if [socket.setdefaulttimeout](https://docs.python.org/3/library/socket.html#socket.setdefaulttimeout) is used.

Reply:
Please see the updated code, I hope I understood correctly.